artio-core/src/main/java/uk/co/real_logic/artio/engine/framer/FixGatewaySessions.java

```diff
@@ -203,6 +205,7 @@ void acquire(
         session.closedResendInterval(gatewaySession.closedResendInterval());
         session.resendRequestChunkSize(gatewaySession.resendRequestChunkSize());
         session.sendRedundantResendRequests(gatewaySession.sendRedundantResendRequests());
+        session.cancelOnDisconnectOption(cancelOnDisconnectOption);
```

Comment:
That doesn't seem right, it will use the default instead of the selected option, plus setting the timeout might be missing too. Would be great to add a test for this.